libimage/layer_tree.go

```diff
@@ -124,6 +128,15 @@ func (r *Runtime) newLayerTreeFromData(images []*Image, layers []storage.Layer)
 		topLayer := img.TopLayer()
 		if topLayer == "" {
 			tree.emptyImages = append(tree.emptyImages, img)
+			if manifestList, _ := img.IsManifestList(context.TODO()); manifestList {
```

Comment:
Do we want to do this for _every_ layer tree? Some  (e.g. for `Image.Parent`) are one-time and one-purpose only.

(I don’t know whether this would be an option to `layerTree`, or a separate data structure parallel to `layerTree`.)

Reply:
I want @vrothberg to comment on this.  We can develop a different structure and process it separately or rename this structure to a different name. I read the tree as a place to cache processed data.